trpcultivate_phenotypes/tests/src/Kernel/Services/Traits/ValidTraitTest.php

```diff
@@ -236,63 +243,435 @@ public function testTraitsServiceDatabaseExpectations() {
+      foreach($keys as $key => $title) {
+        // Each combo, retrieve assets - trait, method and unit.
+
+        // By string parameter.
+        $asset = $this->service_traits->getTraitAsset($combo[ $title ], $key);
+        $asset_bystring_id = (int) $asset->cvterm_id;
+        $this->assertNotEquals($asset_bystring_id, 0, 'Failed to fetch trait asset' . $key . ' : ' . $combo[ $title ] . ' (by string name parameter).');
+
+        // By id number parameter.
+        $asset = $this->service_traits->getTraitAsset($asset_bystring_id, $key);
+        $asset_byinteger_id = (int) $asset->cvterm_id;
+        $this->assertNotEquals($asset_byinteger_id, 0, 'Failed to fetch trait asset' . $key . ' : ' . $combo[ $title ] . ' (by integer id parameter).');
+
+        // Either cases both should match.
+        $this->assertEquals($asset_bystring_id, $asset_byinteger_id,
+          'Trait id returned by trait asset getter with string and integer parameters do not match.');
+      }
+    }
+
+    // Test get trait using trait name or trait id number as parameter
+    // to getTrait() method.
+    foreach($test_combo as $combo) {
+      $name_key = 'Trait Name';
+
+      // By string parameter.
+      $trait = $this->service_traits->getTrait($combo[ $name_key ]);
+      $trait_id = (int) $trait->cvterm_id;
+      $this->assertNotEquals($trait->cvterm_id, 0, 'Failed to fetch trait ' .  $combo[ $name_key ] . ' (by trait name parameter).');
+
+      // By id number parameter.
+      $trait = $this->service_traits->getTrait($trait->cvterm_id);
+      $this->assertNotEquals($trait->cvterm_id, 0, 'Failed to fetch trait ' .  $combo[ $name_key ] . ' (by trait id parameter).');
+
+      // Either cases both should match.
+      $this->assertEquals($trait_id, (int) $trait->cvterm_id,
+        'Trait id returned by trait getter with string and integer parameters do not match.');
+    }
+
+    // Test get trait method using trait name or trait id as parameter
+    // to getTraitMethod() method.
+
+    // Based on the summary of traits assets above, test the following.
+    // 1. A Trait has 3 methods - A, B, C Method.
+    // 2. C Trait has 1 method - D Method.
+    $a_trait_methods_byname = $this->service_traits->getTraitMethod('A Trait');
+    $a_trait = $this->service_traits->getTraitAsset('A Trait', 'trait');
+    $a_trait_methods_byid   = $this->service_traits->getTraitMethod($a_trait->cvterm_id);
+
+    // Assert that in both cases, the returned set of methods was the same.
+    // Then proceed to assert that methods returned are the expected methods.
+    $this->assertEquals($a_trait_methods_byname, $a_trait_methods_byid,
+      'A Trait methods returned by methods getter with name and id as parameter do not match.');
+
+    // 3 methods in the set?
+    $methods_count = count($a_trait_methods_byid);
+    $this->assertEquals($methods_count, 3, 'A Trait methods returned by methods getter does not match expected count (3).');
```

Comment:
It is better to use `assertCount` here. That way if it fails the output shows the array that did not have 3 elements which is more informative.